### PR TITLE
 Bug 1412836 - push_apk.py: expose --do-not-contact-google-play

### PR DESCRIPTION
**Pull request by JohanLorenzo**

### CODE REVIEW

File: mozapkpublisher/common/googleplay.py [hunk 1]
```diff
@@ -36,11 +39,14 @@ def add_general_google_play_arguments(parser):
...
+                        help='The p12 authentication file', required=True)
 
-    parser.add_argument('--dry-run', action='store_true',
-                        help='''Perform every operation of the transation, except committing. No data will be
-stored on Google Play. Use this option if you want to test the script with the same data more than once.''')
+    parser.add_argument('--commit', action='store_true',
+                        help='Commit changes onto Google Play. This action cannot be reverted.')
+    parser.add_argument('--do-not-contact-google-play', action='store_false', dest='contact_google_play',
+                        help='''Prevent any request to reach Google Play. Use this option if you want to run the script
+without any valid credentials nor valid APKs. In fact, Google Play may error out at the first invalid sent piece of data.
```
Comment (MihaiTabara) [inline mozapkpublisher/common/googleplay.py]: tiny nit: I think "... first invalid piece of data sent" sounds better.

File: mozapkpublisher/test/test_push_apk.py [hunk 2]
```diff
@@ -233,6 +228,20 @@ def test_create_or_update_whats_new(edit_service_mock, monkeypatch):
...
+def test_do_not_contact_google_play_flag_does_not_request_google_play(monkeypatch):
+    monkeypatch.setattr(apk, 'check_if_apk_is_multilocale', lambda _: None)
+    monkeypatch.setattr(apk, 'check_if_apk_has_claimed_architecture', lambda _, __: None)
+    set_translations_per_google_play_locale_code(monkeypatch)
+
+    config = copy(VALID_CONFIG)
+    config['do_not_contact_google_play'] = True
+
+    PushAPK(config).run()
+    # Checks are done by the fact that Google Play doesn't error out. In fact, we
```
Comment (MihaiTabara) [inline mozapkpublisher/test/test_push_apk.py]: 👍 for the comment!

File: mozapkpublisher/common/googleplay.py [hunk 3]
```diff
@@ -16,8 +16,11 @@
 import httplib2
 import logging
 
-from oauth2client.service_account import ServiceAccountCredentials
 from apiclient.discovery import build
+from oauth2client.service_account import ServiceAccountCredentials
+# XXX This definitely looks weird in a production module. However, this is useful for
```
[the remark this answers is not in the document]
Comment (JohanLorenzo) [reply mozapkpublisher/common/googleplay.py]: Okay. Done

File: mozapkpublisher/common/googleplay.py (hunk 1 above)
Comment (JohanLorenzo) [reply mozapkpublisher/common/googleplay.py]: Done

**Comment by JohanLorenzo** [comment]
I forgot to cc @catlee about the usage of Mock in production https://github.com/mozilla-releng/mozapkpublisher/pull/48/files#diff-e44ebce783b4b03171a1edb916f7f583R22

**Pull request merged**